source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,356 @@
+less wait time to read and write data, more threads are often unable to
+perform more work and the cost of creating, scheduling, and destroying those
+threads outweigh any gains.
+
+Connection Pool Sizing
+----------------------
+
+The rough guideline for sizing the connection pool is to set the ``minPoolSize``
+to the number of available CPU cores, and to set the ``maxPoolSize`` to double
+the number of CPU cores plus spindles (if using HDDs).
+
+.. example::
+
+   Your application has 4 available cores, and the servers hosting your database
+   have SSDs. Beginning with a ``minPoolSize`` of 4 and a ``maxPoolSize`` of 8
+   is a good starting point for connection pool size.
+
+
+
+.. _connection-storms-java:
+
+Connection Storms
+-----------------
+
+A **connection storm** is an event in which application servers
+start an increasing number of connection requests, but the host server
+CPU is unable to schedule them.
+
+.. example::
+
+   Consider an application that uses MongoDB, and is served from an
+   auto scaling service. A sudden spike in traffic could cause the scaling
+   service to stand up many new application instances. This sudden burst
+   in application instances attempting to establish connections to MongoDB
+   is one example of a connection storm.
+
+Common Causes
++++++++++++++
+
+Common causes for connection storms in MongoDB include:
+
+- Incorrect application behavior
+- Network latency
+- Insufficient hardware for the workload
+- Failover events
+
+In MongoDB, connection storms can have a multiple effects. For example:
+
+- The high volume of connection requests leads to latency in processing.
+- Replication connections between nodes may be blocked or slowed, so replication
+  operations cannot proceed.
+- Administrative connections to the database are blocked, preventing
+  a user from logging in.
+- Monitoring connections to the database are blocked, degrading visibility
+  into the problem.
+- CPU or memory oversubscription due to connection management can lead
+  to a ``mongod`` crashing and restarting. This extra workload on the server
+  can be exclusively attributed to connection handling.
+- In Atlas, connection requests can exceed instance class limits, causing
+  refusal of new connections.
+
+Avoid Connection Storms
+-----------------------
+
+A combination of settings and methods can help to mitigate connection storms.
+These include:
+
+- **maxConnecting**: The maximum number of connections to each server that
+  the driver will concurrently establish before making them available in
+  the connection pool.
+
+- **maxPoolSize**: The maximum number of connections in the connection pool.
+
+- **maxTime()**: The *cumulative* time limit for processing operations on
+  a MongoDB server.
+
+**maxConnecting**
++++++++++++++++++
+
+``maxConnecting`` specifies the maximum number of connections to each server that
+  the driver will concurrently establish before making them available in
```

Comment:
Indentation issue on 284, 285

This top line is [rendering as a bold heading](https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-19628/fundamentals/connection/tuning/#maxconnecting) because of the indents below